src/DominantJuice.sol

```diff
@@ -0,0 +1,396 @@
+        //paymentToken = paymentTerminal.token();
+        (JBFundingCycle memory cycleData,) = controller.currentFundingCycleOf(projectId);
+        cycleExpiryDate = cycleData.start + cycleData.duration;
+    }
+
+    /**
+     * @notice Project creator calls this function to deposit refund bonus before funding cycle start
+     * to increase potential pledger confidence.
+     * @dev Can only be called by the contract owner / project creator.
+     */
+    function depositRefundBonus(uint256 refundBonusAmount) external payable onlyOwner {
+        // Reverts if initialize() has not been called yet.
+        if (projectId == 0) revert DataSourceNotInitialized();
+
+        // Reverts if ETH sent does not match input amount.
+        if (msg.value != refundBonusAmount) revert FundsMustMatchInputAmount(refundBonusAmount);
+
+        // Updates contract refund bonus variable.
+        totalRefundBonus = refundBonusAmount;
+
+        emit RefundBonusDeposited(msg.sender, totalRefundBonus);
+    }
+
+    /// @notice This function gets called when the project receives a payment.
+    /// @dev Part of IJBFundingCycleDataSource.
+    /// @dev This implementation just sets this contract up to receive a `didPay` call.
+    /// @param _data The Juicebox standard project payment data. See https://docs.juicebox.money/dev/api/data-structures/jbpayparamsdata/.
+    /// @return weight The weight that project tokens should get minted relative to. This is useful for optionally customizing how many tokens are issued per payment.
+    /// @return memo A memo to be forwarded to the event. Useful for describing any new actions that are being taken.
+    /// @return delegateAllocations Amount to be sent to delegates instead of adding to local balance. Useful for auto-routing funds from a treasury as payment come in.
+    function payParams(JBPayParamsData calldata _data)
+        external
+        virtual
+        override
+        returns (uint256 weight, string memory memo, JBPayDelegateAllocation3_1_1[] memory delegateAllocations)
+    {
+        // The if statements are the only changes from the payParams() template.
+        // This is also not a payable function, so calls with msg.value will revert.
+        if (getBalance() == 0) revert DataSourceNotInitialized();
+        if (_data.amount.value < minimumPledgeAmount) revert AmountIsBelowMinimumPledge(minimumPledgeAmount);
+
+        // Forward the default weight received from the protocol.
+        weight = _data.weight;
+        // Forward the default memo received from the payer.
+        memo = _data.memo;
+        // Add `this` contract as a Pay Delegate so that it receives a `didPay` call. Don't send any funds to the delegate (keep all funds in the treasury).
+        delegateAllocations = new JBPayDelegateAllocation3_1_1[](1);
+        delegateAllocations[0] = JBPayDelegateAllocation3_1_1(this, 0, "");
+
+        return (weight, memo, delegateAllocations);
```

Comment:
you're using the returns syntax with names in the function signature, so you don't technically need the return here

Reply:
I believe I did that while testing. But yeah, I can remove.